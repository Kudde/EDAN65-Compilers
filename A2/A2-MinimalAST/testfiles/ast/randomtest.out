Program
  List
    Fun
      TypeDecl Type="int"
      IdDecl Decl="gcd1"
      Param
        List
          ParamDecl
            TypeDecl Type="int"
            IdDecl Decl="a"
          ParamDecl
            TypeDecl Type="int"
            IdDecl Decl="b"
      Block
        List
          WhileStmt
            NotEqual
              ID ID="a"
              ID ID="b"
            Block
              List
                IfStmt
                  Greater
                    ID ID="a"
                    ID ID="b"
                  Block
                    List
                      AssignStmt
                        ID ID="a"
                        Sub
                          ID ID="a"
                          ID ID="b"
                  Opt
                    Block
                      List
                        AssignStmt
                          ID ID="b"
                          Sub
                            ID ID="b"
                            ID ID="a"
                DeclStmt
                  TypeDecl Type="int"
                  IdDecl Decl="k"
                  Opt
                    Literal NUMERAL="2"
          ReturnStmt
            ID ID="a"
    Fun
      TypeDecl Type="int"
      IdDecl Decl="gcd2"
      Param
        List
          ParamDecl
            TypeDecl Type="int"
            IdDecl Decl="a"
          ParamDecl
            TypeDecl Type="int"
            IdDecl Decl="b"
      Block
        List
          IfStmt
            Equal
              ID ID="b"
              Literal NUMERAL="0"
            Block
              List
                ReturnStmt
                  ID ID="a"
            Opt
          ReturnStmt
            FunCall ID="gcd2"
              List
                ID ID="b"
                Mod
                  ID ID="a"
                  ID ID="b"
