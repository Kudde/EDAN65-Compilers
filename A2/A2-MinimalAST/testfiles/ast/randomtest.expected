Program
  List
    Fun
      TypeDecl Type="int"
      IdDecl Decl="gcd1"
      Param
        List
          ParamDecl
            TypeDecl Type="int"
            IdDecl Decl="a"
          ParamDecl
            TypeDecl Type="int"
            IdDecl Decl="b"
      Block
        List
          WhileStmt
            NotEqual
              ID ID="a"
              ID ID="b"
            Block
              List
                IfStmt
                  Greater
                    ID ID="a"
                    ID ID="b"
                  Block
                    List
                      AssignStmt
                        ID ID="a"
                        Sub
                          ID ID="a"
                          ID ID="b"
                  Opt
                    Block
                      List
                        AssignStmt
                          ID ID="b"
                          Sub
                            ID ID="b"
                            ID ID="a"
                DeclStmt
                  TypeDecl Type="int"
                  IdDecl Decl="k"
                  Opt
                    Literal NUMERAL="2"
          ReturnStmt
            ID ID="a"
